static/src/javascripts/projects/common/modules/commercial/creatives/hosted-thrasher.js

```diff
@@ -0,0 +1,41 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/template',
+    'common/modules/commercial/hosted-video',
+    'common/modules/commercial/creatives/add-tracking-pixel',
+    'text!common/views/commercial/creatives/hosted-thrasher.html'
+], function (
+    fastdom,
+    template,
+    hostedVideo,
+    addTrackingPixel,
+    hostedVideoStr
+) {
+    var hostedVideoTpl;
+
+    var HostedThrasher = function ($adSlot, params) {
+        this.$adSlot = $adSlot;
+        this.params = params;
+    };
+
+    HostedThrasher.prototype.create = function () {
+        if (!hostedVideoTpl) {
+            hostedVideoTpl = template(hostedVideoStr);
+        }
+
+        var domPromise = fastdom.write(function () {
+            this.$adSlot.append(hostedVideoTpl({ data: this.params}));
+        }.bind(this));
```

Comment:
Yeah, this isn't really something I think needs to be fixed here and now. It's just a bit of a bee in my bonnet - my rationale has always been that as soon as a symbol is public, when I maintain the code I now have to worry about whether anything is using it downstream.

Might be worth discussing over on dotcom-future, as James says.